msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -750,6 +784,43 @@ public void acquireToken(@NonNull final Activity activity,
         );
     }
 
+    /**
+     * Acquire token interactively, will pop-up webUI. Interactive flow will skip the cache lookup.
+     * Default value for {@link UiBehavior} is {@link UiBehavior#SELECT_ACCOUNT}.
+     *
+     * @param activity  Non-null {@link Activity} that will be used as the parent activity for launching the {@link AuthenticationActivity}.
+     *                  All the apps doing interactive request are required to call the
+     *                  {@link PublicClientApplication#handleInteractiveRequestRedirect(int, int, Intent)} within the calling
+     *                  activity {@link Activity#onActivityResult(int, int, Intent)}.
+     * @param resource  The non-null string of resource identifier.
+     * @param loginHint Optional. If provided, will be used as the query parameter sent for authenticating the user,
+     *                  which will have the UPN pre-populated.
+     * @param callback  The Non-null {@link AuthenticationCallback} to receive the result back.
+     *                  1) If user cancels the flow by pressing the device back button, the result will be sent
+     *                  back via {@link AuthenticationCallback#onCancel()}.
+     *                  2) If the sdk successfully receives the token back, result will be sent back via
+     *                  {@link AuthenticationCallback#onSuccess(IAuthenticationResult)}
+     *                  3) All the other errors will be sent back via
+     *                  {@link AuthenticationCallback#onError(MsalException)}.
+     */
+    public void acquireToken(@NonNull final Activity activity,
```

Comment:
I don't think we need to add resourceId as a separate parameter and introduce a whole new set of overloads.  I don't expect many users to use these overloads... instead I suspect that most users will prefer the newly added overload that accepts the AcquireTokenParameters object.  I'd just add resource id to that object and not introduce additional overloads to MSAL

Reply:
First, yes, agree that we should not add new set of overloads. Second, instead of adding new parameters, I think we should add different initialization methods of AcquireTokenParameter and AcquireTokenSilentParameter, like initWithScope() and initWithResource(). And let the developer to choose which to use. As at our end, there is no way (at lease haven't find a perfect one) to distinguish between resource string/id and scopes.